prog/download_boot_image.rb

```diff
@@ -34,23 +34,27 @@ def url
           "postgres17" => "raw",
           "ubuntu" => "img",
           "almalinux" => "qcow2",
+          "debian" => "raw",
           "ai" => "raw"
         }
         image_family = image_name.split("-").first
         suffix = suffixes.fetch(image_family, nil)
         arch = image_name.start_with?("ai-model") ? "-" : "-#{vm_host.arch}-"
         blob_storage_client.get_presigned_url("GET", Config.ubicloud_images_bucket_name, "#{image_name}#{arch}#{version}.#{suffix}", 60 * 60).to_s
       elsif image_name == "ubuntu-noble"
-        arch = (vm_host.arch == "x64") ? "amd64" : "arm64"
+        arch = vm_host.render_arch(arm64: "arm64", x64: "amd64")
         "https://cloud-images.ubuntu.com/releases/noble/release-#{version}/ubuntu-24.04-server-cloudimg-#{arch}.img"
       elsif image_name == "ubuntu-jammy"
-        arch = (vm_host.arch == "x64") ? "amd64" : "arm64"
+        arch = vm_host.render_arch(arm64: "arm64", x64: "amd64")
         "https://cloud-images.ubuntu.com/releases/jammy/release-#{version}/ubuntu-22.04-server-cloudimg-#{arch}.img"
+      elsif image_name == "debian-12"
+        arch = vm_host.render_arch(arm64: "arm64", x64: "amd64")
+        "https://cloud.debian.org/images/cloud/bookworm/#{version}/debian-12-genericcloud-#{arch}-#{version}.raw"
```

Comment:
Since E2E tests and production download images from the internal minio, should I put it to minio if you don't plan to change the version

Reply:
I don't plan to change it, please do feel free.

Reply:
I'm not sure what our past practice is, but I would think we would want to use the official source as the default, and support overriding via environment variables or in `.env.rb`.  Our hosted version of Ubicloud would want to use the internal minio to improve performance and reduce external bandwidth, but we wouldn't want that used by someone hosting their own version of Ubicloud (I assume).

Reply:
yeah, it is the default, but our production and E2E environment is flagged into the accelerated variant. I develop using the source as-is..

Reply:
btw @pykello
 
```
    # YYY: In future all images should be checked for sha256 sum, so the nil
    # default will be removed.
    hashes.fetch([image_name, vm_host.arch, version], nil)
```

is the future now? I noticed in development that my thing worked even though I hadn't encoded a hash yet and was surprised.

Reply:
It decides here: https://github.com/ubicloud/ubicloud/blob/4dc5c281c9a0c93ad551739d015b91f4aaa8e903/prog/download_boot_image.rb#L17. We may convert it to an environment variable flag.

I use this adhoc workflow to upload new versions from official sources to Minio: https://github.com/ubicloud/runner-images/blob/default-images/.github/workflows/ubicloud-image.yml

<img width="541" alt="Screenshot 2024-10-27 at 20 10 46" src="https://github.com/user-attachments/assets/f157202a-d948-41ae-bcd9-94e97ac95629">

Reply:
Here we go. I've verified the checksums and they match
<img width="1166" alt="Screenshot 2024-10-27 at 20 52 14" src="https://github.com/user-attachments/assets/9ea0db54-8191-4c0a-9ab0-2e3f7d5bbf0d">